assets/js/theme/common/utils/translations-utils.js

```diff
@@ -0,0 +1,22 @@
+const TRANSLATIONS = 'translations';
+const convertToDictionary = jsonStr => JSON.parse(jsonStr);
+const isTranslationDictionaryNotEmpty = (dictionary) => !!Object.keys(dictionary[TRANSLATIONS]).length;
+const chooseActiveDictionary = (...dictionaryJsonList) => dictionaryJsonList.map(convertToDictionary).find(isTranslationDictionaryNotEmpty);
```

Comment:
Small proposal - maybe will be better to use for loop here. In case of removing map we need to parse object string twice - firstly in check and then to transform object string returned from find. Also native for is the fastest loop as I know

Reply:
thanks, guys. Yeap, if we just include  parse into find function call, the last one will return jsonString.And we will need to do parse one more time. I've opted for "for loop" since it helps us to avoid extra parsing and it's fast in terms of performance. Appreciate your suggestions.